lib/models/addon.js

```diff
@@ -515,10 +515,15 @@ let addonProto = {
       trees.push(this.jshintAddonTree());
     }
 
-    let mergedTreesForType = mergeTrees(trees, {
-      overwrite: true,
-      annotation: `Addon#treeFor (${this.name} - ${treeType})`,
-    });
+    let mergedTreesForType;
+    if (trees.length !== 0) {
```

Comment:
The custom merge trees that we use (lib/broccoli/merge-trees.js) should already be taking care of this, can you check into why it isn't?

Reply:
But `EMPTY_MERGE_TREE` is truthy, is it not?